TimeStepping.pyx

```diff
@@ -187,6 +295,83 @@ cdef class TimeStepping:
                 self.t += self.dt
         return
 
+    cpdef update_pressure_fourth(self, Grid.Grid Gr, PrognosticVariables.PrognosticVariables PV, DiagnosticVariables.DiagnosticVariables DV):
+        cdef:
+            Py_ssize_t i
+            Py_ssize_t press_shift = DV.get_varshift(Gr, 'perturbation_pressure_potential')
+
+            Py_ssize_t whor_shift = DV.get_varshift(Gr, 'wBudget_removeHorAve')
+            Py_ssize_t whor_rk0_shift = DV.get_varshift(Gr, 'wBudget_removeHorAve_RK0')
+            Py_ssize_t whor_rk1_shift = DV.get_varshift(Gr, 'wBudget_removeHorAve_RK1')
+            Py_ssize_t whor_rk2_shift = DV.get_varshift(Gr, 'wBudget_removeHorAve_RK2')
+            Py_ssize_t whor_rk3_shift = DV.get_varshift(Gr, 'wBudget_removeHorAve_RK3')
+            Py_ssize_t whor_rk4_shift = DV.get_varshift(Gr, 'wBudget_removeHorAve_RK4')
+
+            Py_ssize_t wpress_shift = DV.get_varshift(Gr, 'wBudget_PressureGradient')
+            Py_ssize_t wpress_rk0_shift = DV.get_varshift(Gr, 'wBudget_PressureGradient_RK0')
+            Py_ssize_t wpress_rk1_shift = DV.get_varshift(Gr, 'wBudget_PressureGradient_RK1')
+            Py_ssize_t wpress_rk2_shift = DV.get_varshift(Gr, 'wBudget_PressureGradient_RK2')
+            Py_ssize_t wpress_rk3_shift = DV.get_varshift(Gr, 'wBudget_PressureGradient_RK3')
+            Py_ssize_t wpress_rk4_shift = DV.get_varshift(Gr, 'wBudget_PressureGradient_RK4')
+
+            double a1, a2, a3
+
+        a1 = 0.386708617503269
+        a2 = 0.096059710526147+a1*0.821920045606868
+        a3 = 0.517231671970585+a2*0.379898148511597
+        a4 = 0.555629506348765*a3
+
+        with nogil:
+            if self.rk_step == 0:
+                if self.t % self.statIOdt == 0.0:
```

Comment:
Is this condition necessarily met? I do not know how this behaves when the output timestep is not a multiple of the prognostic timestep (which is conditioned by the CFL).